code/datums/components/burning.dm

```diff
@@ -63,6 +71,37 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
 
 	examine_list += span_danger("[source.p_theyre(TRUE)] burning!")
 
+/// Handles searing the hand of anyone who tries to touch this without protection.
+/datum/component/burning/proc/on_attack_hand(atom/source, mob/living/carbon/user)
+	SIGNAL_HANDLER
+
+	// attack_hand() behavior should only apply to items, for now
+	if(!isitem(source))
+		return NONE
+
+	var/can_handle_hot = FALSE
+	if(!istype(user))
+		can_handle_hot = TRUE
```

Comment:
I don't like implicit istypes. you should be explicit about this IMO, even if it requires an extra indentation

Reply:
...fine?